clang/lib/Headers/__stddef_offsetof.h

```diff
@@ -7,6 +7,11 @@
  *===-----------------------------------------------------------------------===
  */
 
-#ifndef offsetof
+/*
+ * When -fbuiltin-headers-in-system-modules is set this is a non-modular header
+ * and needs to behave as if it was textual.
+ */
+#if !defined(offsetof) ||                                                      \
+    (__has_feature(modules) && !__building_module(_Builtin_stddef))
```

Comment:
What should happen when `offsetof` is defined and we are building non-`_Builtin_stddef` module?

Reply:
This header shouldn't be seen at all, and `_Builtin_stddef.offset` should instead be imported/made visible

Reply:
I was thinking about `offsetof` being defined in some header, not in the intended system one. And then we [transitively] include "\_\_stddef_offsetof.h". Does it mean that in this case "\_\_stddef_offsetof.h" is processed only when building module "_Builtin_stddef"?

Reply:
Oh I see what you mean. When the Darwin module builds, it will build <stddef.h> which will include <__stddef_offsetof.h>. It will bypass the header guard and (re)define `offsetof`. The Darwin module will also build <sys/_types/_offsetof.h>, so which one gets used is undefined, but the Darwin module will be the single definer of `offsetof`. It's a little weird, but this actually matches the LLVM 17 behavior.